File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/validation/ValidDateRange.java
package com.github.arhor.simple.expense.tracker.validation;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import javax.validation.Constraint;
import javax.validation.Payload;

import static java.lang.annotation.ElementType.PARAMETER;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target({PARAMETER})
@Retention(RUNTIME)
@Documented
@Constraint(validatedBy = {})
public @interface ValidDateRange {

    String message() default "{com.acme.constraint.OrderNumber.message}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/validation/ValidDateRange.kt
package com.github.arhor.simple.expense.tracker.validation

import javax.validation.Constraint
import javax.validation.Payload
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.VALUE_PARAMETER
import kotlin.reflect.KClass

@Target(VALUE_PARAMETER)
@Retention(RUNTIME)
@MustBeDocumented
@Constraint(
    validatedBy = [
        /* @see resources/META-INF/services/javax.validation.ConstraintValidator */
    ]
)
annotation class ValidDateRange(
    val message: String = "{com.acme.constraint.OrderNumber.message}",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<in Payload>> = [],
)
